pkgs/development/python-modules/torch/default.nix

```diff
@@ -184,6 +167,25 @@ in buildPythonPackage rec {
       --replace "set(ROCM_PATH \$ENV{ROCM_PATH})" \
         "set(ROCM_PATH \$ENV{ROCM_PATH})''\nset(ROCM_VERSION ${lib.concatStrings (lib.intersperse "0" (lib.splitString "." hip.version))})"
   ''
+  # Detection of NCCL version doesn't work particularly well when using the static binary.
+  + lib.optionalString cudaSupport ''
+    substituteInPlace cmake/Modules/FindNCCL.cmake \
+      --replace \
+        'message(FATAL_ERROR "Found NCCL header version and library version' \
+        'message(WARNING "Found NCCL header version and library version'
+  ''
+  # Propagate CUPTI to Kineto by overriding the search path with environment variables.
+  # https://github.com/pytorch/pytorch/blob/774c8229795f5fc9f8cc1ea39011fa6468b638a3/cmake/Dependencies.cmake#L1907-L1919
+  # Match only on the text following the dollar sign to avoid weird substitution issues.
+  + lib.optionalString cudaSupport ''
+    substituteInPlace cmake/Dependencies.cmake \
+      --replace \
+        "{CUDA_SOURCE_DIR}/extras/CUPTI/include" \
+        "ENV{CUPTI_INCLUDE_DIR}" \
+      --replace \
+        "{CUDA_SOURCE_DIR}/extras/CUPTI/lib64" \
+        "ENV{CUPTI_LIBRARY_DIR}"
+  ''
```

Comment:
I opened https://github.com/pytorch/pytorch/pull/108847 to address this upstream. Perhaps we could pull that via a patch or just link to it in the code comment for

Reply:
@samuela I added a comment and the patch, but the patch doesn't work when using `ENV` instead of `$ENV`.